src/main/java/com/flagsmith/FlagsmithCachedAPIWrapper.java

```diff
@@ -56,4 +61,30 @@ public FlagsAndTraits identifyUserWithTraits(FeatureUser user, List<Trait> trait
   public FlagsmithCache getCache() {
     return cache;
   }
+
+  private FlagsAndTraits getCachedFlagsIfTraitsMatch(FeatureUser user, List<Trait> traitsToMatch) {
+    final FlagsAndTraits flagsAndTraits = cache.getCache().getIfPresent(user.getIdentifier());
+    if (flagsAndTraits == null) {
+      // cache doesnt exist for this user
+      return null;
+    } else if (flagsAndTraits.getTraits() != null) {
+
+      final boolean allTraitsFound =
+          traitsToMatch == null ||
+          traitsToMatch.isEmpty() ||
+          traitsToMatch.stream().allMatch(t -> {
+            final Trait newTrait = new Trait(null, t.getKey(), t.getValue());
```

Comment:
I'm sure I've asked this before and you gave me a good answer but I'm afraid I've forgotten... Is there a reason this line isn't: 

```java
final Trait newTrait = new Trait(user.getIdentifier(), t.getKey(), t.getValue());
```

Reply:
Hi @matthewelwell this is because the SDK is being tested against the Flagsmith API and the API does not populate this field. If you would like to change the java SDK, you will need to update the Flagsmith API first.